packages/smart-wallet/src/walletFactory.js

```diff
@@ -192,7 +194,9 @@ export const prepare = async (zcf, privateArgs, baggage) => {
         const walletP =
           !walletsByAddress.has(address) && walletReviver
             ? // this will call provideSmartWallet which will update `walletsByAddress` for next time
-              E(walletReviver).reviveWallet(address)
+              E(walletReviver)
+                .reviveWallet(address)
+                .then(([wallet, _isNew]) => wallet)
```

Comment:
```suggestion
              E.when(
                E(walletReviver).reviveWallet(address),
                ([wallet, _isNew]) => wallet)
```

The `.then` you've written is on the result of an `E` and so is safe from any concern about reentry. The reason to shift to `E.when` anyway is so that deep stacks, when they happen to be enabled, will see the causality chain.

This occurs with several other `.then`s in this PR (and probably many more elsewhere)